loky/backend/context.py

```diff
@@ -154,16 +154,46 @@ def _cpu_count_cgroup(os_cpu_count):
             return os_cpu_count
 
 
-def _cpu_count_user(os_cpu_count):
-    """Number of user defined available CPUs"""
+def _cpu_count_affinity(os_cpu_count):
     # Number of available CPUs given affinity settings
-    cpu_count_affinity = os_cpu_count
     if hasattr(os, 'sched_getaffinity'):
         try:
-            cpu_count_affinity = len(os.sched_getaffinity(0))
+            return len(os.sched_getaffinity(0))
         except NotImplementedError:
             pass
 
+    # On PyPy and possibly other platforms, os.sched_getaffinity does not exist
+    # or raises NotImplementedError, let's try with the psutil if installed.
+    try:
+        import psutil
+
+        p = psutil.Process()
+        if hasattr(p, "cpu_affinity"):
+            return len(p.cpu_affinity())
+
+    except ImportError:  # pragma: no cover
```

Comment:
Covering this case would require configuring a second pypy worker with a different version (e.g. PyPy 3.8) but I don't think it's worth the extra CI resource usage just to test for a warning. I checked locally that this warning works.